### PR TITLE
Use ranger's `choosedir` option and do some cleanups.

### PR DESCRIPTION
**Pull request by infokiller**

### CODE REVIEW

**Comment by rafaqz** [comment]
Hey thanks for  cleaning up my mediocre vimscript. One question (for my understanding) why `abort` everywhere?

**Comment by infokiller** [comment]
I'm far from an expert in vimscript, but I'm following [Google's guideline](https://google.github.io/styleguide/vimscriptguide.xml?showone=Functions#Functions)

File: plugin/ranger.vim [hunk 1]
```diff
@@ -17,19 +17,24 @@ function! s:RangerMagic(path) " {{{
 
   " Create a new temporary file name
   let g:ranger_tempfile = tempname()
-  let opts = ' --choosefiles=' . shellescape(g:ranger_tempfile) . ' ' . shellescape(a:path)
+  if g:ranger_cd_mode
+    let l:choose_arg = 'choosedir'
+  else
+    let l:choose_arg = 'choosefiles'
+  endif
+  let l:opts =  printf('--%s=%s %s', l:choose_arg, g:ranger_tempfile, shellescape(a:path))
```
Comment (rafaqz) [inline plugin/ranger.vim]: Why remove the shellescape on `g:ranger_tempfile`?

Comment (infokiller) [reply plugin/ranger.vim]: Was done accidentally, now I fixed it.

**Pull request merged**